src/spriteloader/grf.cpp

```diff
@@ -273,7 +274,10 @@ uint8 LoadSpriteV2(SpriteLoader::Sprite *sprite, uint8 file_slot, size_t file_po
 		byte colour = type & SCC_MASK;
 		byte zoom = FioReadByte();
 
-		if (colour != 0 && (load_32bpp ? colour != SCC_PAL : colour == SCC_PAL) && (sprite_type != ST_MAPGEN ? zoom < lengthof(zoom_lvl_map) : zoom == 0)) {
+		if (colour != 0 &&
+			(load_32bpp ? colour != SCC_PAL : colour == SCC_PAL) &&
+			(sprite_type != ST_MAPGEN ? zoom < lengthof(zoom_lvl_map) && zoom_lvl_map[zoom] >= _settings_client.gui.sprite_zoom_min : zoom == 0)
+			) {
```

Comment:
I think you forgot "git add src/table/settings.ini" :)
I can't tell whether changing the setting in-game has immediate effect, or whether it needs a restart.

Reply:
It has immediate effect where all of the zoom settings call a method to revalidate the current zoom. I'm still split on whether to also dirty the entire viewport at this point to force all sprites to reload rather than briefly showing a mixture of 1x/2x/4x until the region they're in becomes dirty.

settings.ini needs fixing :)

Reply:
After further playing with the setting I think it's a bit weird to change a setting and have it apply inconsistently until the screen is scrolled or some animation happens, so I enforced a screen refresh.